lab_1_keywords_tfidf/main.py

```diff
@@ -2,158 +2,256 @@
 Lab 1
 Extract keywords based on frequency related metrics
 """
-from typing import Optional, Union
+import math
+from typing import Optional, Union, Any
+import string
 
 
-def clean_and_tokenize(text: str) -> Optional[list[str]]:
+def correct_list(list1: list, type1: Any, empty: bool) -> bool:
     """
-    Removes punctuation, casts to lowercase, splits into tokens
-
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
+    Checks that type of 'list1' is list and verifies the contents of 'list1' with the type that the user specifies
     """
-    pass
+    if isinstance(list1, list):
+        if list1 == [] and empty is False:
+            return False
+        for index in list1:
+            if isinstance(index, type1) is False:
+                return False
+        return True
+    return False
 
 
-def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list[str]]:
+def correct_dict(dictionary: dict, type1: Any, type2: Any, empty: bool) -> bool:
     """
-    Excludes stop words from the token sequence
-
-    Parameters:
-    tokens (List[str]): Original token sequence
-    stop_words (List[str]: Tokens to exclude
-
-    Returns:
-    List[str]: Token sequence that does not include stop words
-
-    In case of corrupt input arguments, None is returned
+    Checks that type of 'dictionary' is dict and verifies the contents of 'dictionary' with the type that
+    the user specifies
     """
-    pass
+    if isinstance(dictionary, dict):
+        if dictionary == {} and empty is False:
+            return False
+        for key, value in dictionary.items():
+            if not isinstance(key, type1) or not isinstance(value, (int, type2)) or isinstance(value, bool):
+                return False
+        return True
+    return False
 
 
-def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
+def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
-    Composes a frequency dictionary from the token sequence
+        Removes punctuation, casts to lowercase, splits into tokens
```

Comment:
Why are docstrings changed? Please return them to their original condition

Reply:
It's not my changes

Reply:
But GitHub shows that you changed it. Anyway - return every docstring to the original view